app/services/samples/metadata/update_service.rb

```diff
@@ -0,0 +1,101 @@
+# frozen_string_literal: true
+
+module Samples
+  module Metadata
+    # Service used to Update Samples::Metadata
+    class UpdateService < BaseService
+      SampleMetadataUpdateError = Class.new(StandardError)
+      attr_accessor :sample, :metadata, :analysis_id
+
+      def initialize(project, sample, user = nil, params = {})
+        super(user, params)
+        @project = project
+        @sample = sample
+        @metadata = params['metadata']
+        @analysis_id = params['analysis_id']
+      end
+
+      def execute
+        authorize! sample.project, to: :update_sample?
+
+        validate_sample_in_project
+
+        validate_metadata_param
+
+        transform_metadata_keys
+
+        metadata_fields_update_status = metadata_modification
+
+        @sample.save
+
+        fields_not_updated(metadata_fields_update_status)
+        metadata_fields_update_status
+      rescue Samples::Metadata::UpdateService::SampleMetadataUpdateError => e
+        @sample.errors.add(:base, e.message)
+        metadata_fields_update_status || { updated: [], not_updated: [] }
```

Comment:
You should be able to just return `metadata_fields_update_status` here as well. In the controller we can always check for a nil value as well

Reply:
Updated in 5ab62ea9e785867b0a42761d594ba86b2f4f0a82